packages/liveblocks-core/src/immutable.ts

```diff
@@ -304,17 +308,13 @@ function legacy_patchImmutableNode<S extends Json>(
   if (pathItem === undefined) {
     switch (update.type) {
       case "LiveObject": {
-        if (
-          state === null ||
-          typeof state !== "object" ||
-          Array.isArray(state)
-        ) {
+        if (!isJsonObject(state)) {
           throw new Error(
             "Internal: received update on LiveObject but state was not an object"
           );
         }
 
-        const newState = Object.assign({}, state as JsonObject);
+        const newState: JsonObject = Object.assign({}, state);
```

Comment:
FYI: Using `as` is an unsafe construct in TypeScript, telling TypeScript you know it better which is potentially unsafe, especially the value you're casting changes in the future. Here, I'm switching to using an explicit variable annotation to cause an intermediate explicit type check, which has to always match TypeScript's infered types. If this later causes a mismatch, it will be reported. This is safer in the light of future changes than using an `as` construct.